towhee/dag/graph_repr.py

```diff
@@ -40,15 +51,33 @@ def operators(self) -> Dict[str, OperatorRepr]:
     def dataframes(self) -> Dict[str, DataframeRepr]:
         return self._dataframes
 
-    def from_yaml(self, yaml: str):
+    def from_yaml(self, src: str):
         """Import a YAML file describing this graph.
 
         Args:
-            yaml:
+            src:
                 YAML file (pre-loaded as string) to import.
         """
-        # TODO(Chiiizzzy)
-        raise NotImplementedError
+        # load YAML file as a dictionray
+        graph_dict = yaml.safe_load(src)
+
+        # some simple schema checks for YAML file
+        essentials = {'graph', 'operators', 'dataframes'}
+        if not isinstance(graph_dict, dict):
+            raise ValueError('src is not a valid YAML file')
+        if not essentials.issubset(set(graph_dict.keys())):
+            raise ValueError('src cannot descirbe a DAG in Towhee')
+
+        # load name from YAML
+        self._name = graph_dict['graph']['name']
+
+        # load dataframes from YAML
+        for df in graph_dict['dataframes']:
+            self._dataframes[df['name']] = DataframeRepr(df['name'], str(df))
```

Comment:
Are you sure we can use `str(df)` to transform this back into a YAML string? Isn't `df` a nested dictionary?

Reply:
My bad, it loads dataframe from a string representation of a dict. Still, yaml.safe_load can load a dict from it

Reply:
Weird, are you sure the resulting representation is still correct in all cases?

Reply:
I've checked again and printed the result. It seems fine. In which case do you think will go wrong?

Reply:
I am not 100% sure, but the output of `str(df)` is not YAML, there are probably cases where it will not work properly

Reply:
fixed